src/common/transformations/src/transformations/low_precision/mark_dequantization_subgraph.cpp

```diff
@@ -357,3 +358,50 @@ ov::pass::KeepDequantizationPrecision::KeepDequantizationPrecision(const element
     auto m = std::make_shared<Matcher>(multiply_pattern, "KeepDequantizationPrecision");
     this->register_matcher(m, callback);
 }
+
+ov::pass::MarkGatherSubgraph::MarkGatherSubgraph(const element::TypeVector& fp_precisions_to_mark,
+                                                 const element::TypeVector& int_precisions_to_mark) {
+    MATCHER_SCOPE(MarkGatherSubgraph);
+
+    // 1. Data input: FP → (optional Convert) → (input to Gather[0])
+    auto is_fp = [](const Output<Node>& out) {
+        return out.get_element_type().is_real();
+    };
+    auto data_input = wrap_type<op::v0::Constant>(is_fp && check_precision(fp_precisions_to_mark));
```

Comment:
Is `is_fp` required ?
The `fp_precisions_to_mark` should have only FP and will validate Node output?

Same for `is_integral` check

Reply:
`is_fp` and `is_integral` are removed